sky/skylet/attempt_skylet.py

```diff
@@ -0,0 +1,46 @@
+"""Restarts skylet if version does not match"""
+
+import os
+import subprocess
+
+from sky.skylet import constants
+
+
+def restart_skylet(kill_old=False):
+    with open(constants.SKYLET_VERSION_FILE, 'w+') as v_f:
+        v_f.write(constants.SKYLET_VERSION)
+
+    if kill_old:
+        #pylint: disable=subprocess-run-check
+        subprocess.run(
+            'ps aux | grep "sky.skylet.skylet" | grep "python3 -m"'
+            '| awk \'{print $2}\' | xargs kill',
+            shell=True,
+        )
+    subprocess.run(
+        'nohup python3 -m sky.skylet.skylet'
+        ' >> ~/.sky/skylet.log 2>&1 &',
+        shell=True,
+        check=True)
+
+
+proc = subprocess.run(
+    'ps aux | grep "sky.skylet.skylet" | grep "python3 -m"'
+    '| grep -v "grep"',
+    shell=True,
+    capture_output=True,
+    check=True)
+
+running = proc.stdout.decode().strip().replace('\n', '')
+
+if not running:
+    restart_skylet(kill_old=False)
```

Comment:
Do we need this? It seems we will not check the return code for killing the skylet process later, should we just remove these lines?

Reply:
We do need to check if the skylet is running so we can restart it if it somehow died (even if the version file matches but  I agree that if we don't check the return code for killing it we don't have to use `running` as a condition of whether to kill the old one or not)